provider/vsphere/provider.go

```diff
@@ -69,9 +71,39 @@ func (p environProvider) CloudSchema() *jsonschema.Schema {
 
 // Ping tests the connection to the cloud, to verify the endpoint is valid.
 func (p environProvider) Ping(endpoint string) error {
-	// a stub for now until we get a real implementation.
-	_, err := url.Parse(endpoint)
-	return errors.Trace(err)
+	// try to be smart and not punish people for adding or forgetting http
+	var u *url.URL
+	var err error
+	if strings.HasPrefix(strings.ToLower(endpoint), "http") {
+		u, err = url.Parse(endpoint)
+		if err != nil {
+			return errors.New("Invalid endpoint format, please give a full url or IP/hostname.")
+		}
+	} else {
```

Comment:
if we are going to be forgiving then check also that there is no other protocol passed before just prepending one.